src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs

```diff
@@ -27,9 +27,14 @@ namespace Microsoft.AspNetCore.Mvc.Razor.Internal
     /// </summary>
     public class DefaultRoslynCompilationService : ICompilationService
     {
-        private readonly DebugInformationFormat _pdbFormat = SymbolsUtility.SupportsFullPdbGeneration() ?
-            DebugInformationFormat.Pdb :
+        private readonly DebugInformationFormat _pdbFormat =
+#if NET451
+            SymbolsUtility.SupportsFullPdbGeneration() ?
+                DebugInformationFormat.Pdb :
+                DebugInformationFormat.PortablePdb;
+#else
             DebugInformationFormat.PortablePdb;
+#endif
```

Comment:
We end up having to `#ifdef` the usings too. So it ends up about the same number of `#ifdef`s